test/e2e/common_suite_test.go

```diff
@@ -445,3 +445,38 @@ func doTestPodVMwithAnnotationsLargerCPU(t *testing.T, assert CloudAssert) {
 	}
 	newTestCase(t, "PodVMwithAnnotationsLargerCPU", assert, "Failed to Create PodVM with Annotations Larger CPU").withPod(pod).withInstanceTypes(testInstanceTypes).withCustomPodState(v1.PodPending).run()
 }
+
+func doTestServiceAndExtraPod(t *testing.T, assert CloudAssert) {
```

Comment:
Shall we rename the func name to `doTestPodToServiceCommunication`? I guess this is the real purpose of this test case?